File: db4oplugin/db4oplugin/src/com/db4o/eclipse/Db4oPluginActivator.java
package com.db4o.eclipse;

import org.eclipse.jface.resource.*;
import org.eclipse.ui.plugin.*;
import org.osgi.framework.*;

public class Db4oPluginActivator extends AbstractUIPlugin {

	public static final String PLUGIN_ID = "db4oplugin";

	private static Db4oPluginActivator plugin;

	public void start(BundleContext context) throws Exception {
		super.start(context);
		plugin = this;
	}

	public void stop(BundleContext context) throws Exception {
		plugin = null;
		super.stop(context);
	}
		
	public static Db4oPluginActivator getDefault() {
		return plugin;
	}

	public static ImageDescriptor getImageDescriptor(String path) {
		return imageDescriptorFromPlugin(PLUGIN_ID, path);
	}
	
}

File: db4oplugin/db4oplugin/src/com/db4o/eclipse/Db4oPluginActivator.scala
package com.db4o.eclipse;

import org.eclipse.jface.resource._;
import org.eclipse.ui.plugin._;
import org.osgi.framework._;

object Db4oPluginActivator {
	val PLUGIN_ID = "db4oplugin"

 	var plugin: Db4oPluginActivator = null

	def getDefault = plugin

	def getImageDescriptor(path: String) = AbstractUIPlugin.imageDescriptorFromPlugin(PLUGIN_ID, path)
}

class Db4oPluginActivator extends AbstractUIPlugin {

	override def start(context: BundleContext) {
		super.start(context);
		Db4oPluginActivator.plugin = this;
	}

	override def stop(context: BundleContext) {
		Db4oPluginActivator.plugin = null;
		super.stop(context);
	}
}
